<?xml version="1.0"?>
<robot name="Puzzlebot">

  <!-- Base Link -->
  <link name="base_link">
  </link>
  
  <!-- Chassis Joint (Static) -->
  <joint name="chassis_joint" type="fixed">
    <parent link="base_link"/>
    <child link="chassis"/>
    <origin xyz="0.0 0.0 0.05"/>
  </joint>

  <!-- Chassis Link -->
  <link name="chassis">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <mass value="0.6"/>
      <inertia ixx="7.58333333333e-06" ixy="0.0" ixz="0.0" iyy="7.58333333333e-06" iyz="0.0" izz="1.25e-05"/>
    </inertial>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://puzzlebot_sim/models/MCR2_1000_13_Chassis.stl"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="1.57079632679 0 1.57079632679" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://puzzlebot_sim/models/MCR2_1000_13_Chassis.stl"/>
      </geometry>
    </visual>
  </link>

  <!-- Left Wheel Link -->
<link name="wl_link">
  <inertial>
    <origin xyz="0 0 0" />
    <mass value="0.5"/>
    <inertia ixx="0.0001" ixy="0" ixz="0" iyy="0.0001" iyz="0" izz="0.0002"/>
  </inertial>
  <collision>
    <origin xyz="0 0 0" />
    <geometry>
      <mesh filename="package://puzzlebot_sim/models/MCR2_1000_13_Left_Wheel.stl"/>
    </geometry>
  </collision>
  <visual>
    <origin xyz="0 0 0" />
    <geometry>
      <mesh filename="package://puzzlebot_sim/models/MCR2_1000_13_Left_Wheel.stl"/>
    </geometry>
  </visual>
</link>

  <!-- Left Wheel Joint -->
  <joint name="leftWheel" type="continuous">
    <parent link="chassis"/>
    <child link="wl_link"/>
    <origin xyz="0.055 -0.1 0" rpy="0 1.57079632679 1.57079632679"/>  <!-- Yaw de 90 grados -->
    <axis xyz="0 0 1"/>
  </joint>

  <!-- Right Wheel Link -->
<link name="wr_link">
  <inertial>
    <origin xyz="0 0 0" />
    <mass value="0.5"/>
    <inertia ixx="0.0001" ixy="0" ixz="0" iyy="0.0001" iyz="0" izz="0.0002"/>
  </inertial>
  <collision>
    <origin xyz="0 0 0" />
    <geometry>
      <mesh filename="package://puzzlebot_sim/models/MCR2_1000_13_Right_Wheel.stl"/>
    </geometry>
  </collision>
  <visual>
    <origin xyz="0 0 0" />
    <geometry>
      <mesh filename="package://puzzlebot_sim/models/MCR2_1000_13_Right_Wheel.stl"/>
    </geometry>
  </visual>
</link>


  <!-- Right Wheel Joint -->
  <joint name="rightWheel" type="continuous">
    <parent link="chassis"/>
    <child link="wr_link"/>
    <origin xyz="0.055 0.1 0" rpy="0 1.57079632679 1.57079632679"/>  <!-- Yaw ajustado a -Ï€/2 radianes -->
    <axis xyz="0 0 1"/>
  </joint>

</robot>